crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -954,13 +1018,24 @@ impl LogsController {
         });
     }
 
-    fn seal_log(&mut self, log_id: LogId, segment_index: SegmentIndex) {
+    fn seal_log(&mut self, log_id: LogId, segment_index: SegmentIndex, attempt: AttemptsCounter) {
         let tc = task_center().clone();
         let bifrost = self.bifrost.clone();
         let metadata_store_client = self.metadata_store_client.clone();
         let metadata_writer = self.metadata_writer.clone();
+        let retry_policy = self.retry_policy.clone();
+
         self.async_operations.spawn(async move {
             tc.run_in_scope("logs-controller-seal-log", None, async {
+                if attempt > AttemptsCounter::FIRST {
+                    let delay = retry_policy
+                        .iter()
+                        .nth(usize::from(attempt) - 1)
+                        .unwrap_or(FALLBACK_MAX_RETRY_DELAY);
+                    tracing::debug!(?delay, %attempt, %log_id, %segment_index, "Wait before attempting to seal log");
+                    tokio::time::sleep(delay).await;
+                }
```

Comment:
Instead of creating a new retry_policy iterator on every attempt and then finding the right iterator entry, could we pass an `Option<RetryIter<'static>>` along? That way, we capture the attempt and also directly the next delay. `None` would mean no delay.

Reply:
Yeah, that's definitely a better approach.